### DISCUSSION TITLE
F-Droid inclusion

### DISCUSSION THREAD

**Discussion by shuvashish76** [LinkSheet/LinkSheet#593] [General]
### Feature description

Would you consider publishing the app on F-Droid please?

**Comment by 1fexd** [comment]
Yes, I will submit it to F-Droid once it is polished enough.

**Comment by shuvashish76** [comment]
Please keep the issue open or convert it to a discussion.

**Comment by 1fexd** [comment]
Once LinkSheet is polished enough, I'll think about publishing it on F-Droid / other app stores.

**Reply by TPS** [reply]
@IzzySoft Is it reasonable to have current rolling releases [updated in your repo](https://apt.izzysoft.de/fdroid/index/apk/fe.linksheet) while the eventual official F-Droid release continues to bake?

**Reply by IzzySoft** [reply]
What "rolling releases"? The latest release here is from 2023, the latest pre-release from 2024. I do not see any other releases _in this repo here_. The repo holding the "rolling/nightly" ones holds no code. So unfortunately not, sorry: mixing two repos for one app, causes a mess here (while it might work, it will throw a lot of "false alerts" at our end).

What _would_ work: having a "stable pre-release" transferred here every now and then – say, once a quarter or so. While we also prefer not to use pre-releases, exceptions for that are possible (and we could make an exception here).

**Reply by TPS** [reply]
@IzzySoft I'm assuming, then, that [this nightly section from the repo](https://github.com/LinkSheet/LinkSheet/#nightly-builds) is no help:
> ## Nightly builds
>
>Currently, LinkSheet uses a "nightly" "rolling release" model which allows for fast development iteration at the cost of stability; However, since this has proven to work quite well, there usually aren't any issues to be expected.
>
>* Latest build is always available via [nightly.link](https://nightly.link/LinkSheet/LinkSheet/workflows/build-nightly/master/linksheet-nightly.zip)
>
>* Use [Obtainium](https://github.com/ImranR98/Obtainium) with the [nightly](https://github.com/LinkSheet/nightly) repository to auto-download builds

**Reply by IzzySoft** [reply]
As I wrote: the repo must have the APKs at it releases – and the code in its tree. Both at the same time. Not code in repo 1, APKs in repo 2. And yes, I've seen the other repo – which is why I wrote this :wink:

**Reply by TPS** [reply]
@IzzySoft Thanks for looking into it! 🙇🏾‍♂️

**Reply by 1fexd** [reply]
> @IzzySoft Thanks for looking into it! 🙇🏾‍♂️

I'll publish regular builds again eventually, what's the issue with using Obtainium in the mean time?

**Reply by TPS** [reply]
I'm fairly technical, but can't figure it out in a reasonable manner that involves less work for me. Also, yet _another_ app updater just seems… unnecessary.  If you wish to persist w/ current setup, why not publish your _own_ F-Droid repo?

**Reply by IzzySoft** [reply]
> why not publish your own F-Droid repo?

That's what I know several other projects do as well: running their own "nightly" repo – even if they post "regular updates" once a month or more often, available at F-Droid and/or IzzyOnDroid then. This way, whoever wants "bleeding edge", can simply add that "nightly" repo to their client (our clients, Droid-ify and Neo Store, even have such repos pre-configured, so folks only need to "toggle them on").

The "official repos" (F-Droid, IzzyOnDroid) then have the advantage of additional screening, and potentially even [Reproducible Builds](https://izzyondroid.org/about/security/ReproducibleBuilds/), for those who prefer "stable and extra checks" – while yours offers faster and more updates, and "bleeding edge".

**Comment by TPS** [comment]
Old release @ https://apt.izzysoft.de/packages/fe.linksheet .  Is this ready for main F-Droid, @1fexd?

**Comment by ballo** [comment]
I installed it via Obtainium. I see no reason to add F-Droid so long that works well
